spotbugs/src/main/java/edu/umd/cs/findbugs/SwitchHandler.java

```diff
@@ -120,6 +126,45 @@ public SourceLineAnnotation getCurrentSwitchStatement(BytecodeScanningDetector d
                 detector.getClassContext(), detector, details.switchPC, details.switchPC + details.maxOffset - 1);
     }
 
+    public void sawInvokeDynamic(int pc, String methodName) {
+        if ("typeSwitch".equals(methodName)) {
+            typeSwitchPcs.add(pc + 5);
+        }
+    }
+
+    /**
+     * @param opCode The operation code
+     * @param pc The program counter
+     * @return <code>true</code>If this instruction is a cast for a type switch
+     */
+    public boolean isTypeSwitchCaseCheckCast(int opCode, int pc) {
+        if (opCode != Const.CHECKCAST) {
+            return false;
+        }
+
+        // For a type switch each case starts with an aload_2 followed by a checkcast
+        // Since we're on a checkcast look for a switch with a target on the previous PC
+        SwitchDetails switchDetails = findSwitchDetailsByPc(pc - 1);
+
+        if (switchDetails != null) {
```

Comment:
This is essentially "if not...else" which is a double negative. Perhaps refactor to "if null then return false".

Reply:
Yes, I have made the change